src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/common/actions/BaseResourceAllocationAction.java

```diff
@@ -196,6 +201,8 @@ private void doAllocateInTransaction(T stateMachine) {
         } catch (FailsafeException ex) {
             throw ex.getCause();
         }
+
+        checkAllocatedPaths(stateMachine);
```

Comment:
if this check fails will we end up with stalled resources allocated for the flow? since transaction is committed

Reply:
Yes, you are right. Added a method for removing paths.